### DISCUSSION TITLE
CSS not being applied in pages :/

### DISCUSSION THREAD

**Discussion by noethix** [community/community#22495] [New to GitHub]
<p>Hello eveyone,</p>
<p>I am new to github. I’ve been following tutoials etc on how to set up pages. Now, i’m trying to move my codepen HTML and CSS over here. HTML is working, but CSS is not being applied. </p>
<p>I’m obviously doing something wrong. But I can’t figure it out.</p>
<p><a href="https://github.com/noethix/trial" rel="nofollow noopener">https://github.com/noethix/trial</a></p>
<p>Thank you in advance fo any pointers.</p>
<p>Cecile</p>

**Comment by SethClydesdale** [comment]
<p>Hi <a class="mention" href="/noethix">@noethix</a>,</p>
<p>Looking at your page, it appears the issue is the “/” at the beginning of your stylesheet’s source. Change “/css/main.css” to “css/main.css” (<a href="https://github.com/noethix/trial/blob/master/index.html#L5" rel="nofollow noopener">on line 5 of index.html</a>) and it should fix the issue you’re having.</p>
<p>For future reference, if the html file is in the same directory as another folder or file, you don’t need to include a slash at the beginning of the src or href attributes. Also, if you need to access files up one, two, or however many directories, you can use two periods as a substitute. (example below)</p>
<p>say our folders are…</p>
<p>root --&gt; resources --&gt; stylesheet.css</p>
<p>root --&gt; lessons --&gt; lesson-1 --&gt; exercise-1 --&gt; index.html</p>
<p>For our html file to access the stylesheet we could provide the following.</p>
<pre><code>&lt;link rel="stylesheet" href="../../../resources/stylesheet.min.css"&gt;
</code></pre>
<p>Basically two “…” moves up a directory and since index.html was three directories down, we moved up three times until we were in the root thus able to access the resources folder. You can also reference the current directory with “./”.</p>
<p>If any questions, let us know.</p>

**Reply by beffyd** [reply]
THANKYOU

**Reply by ShivamQmr** [reply]
https://github.com/Shivam2080/rajuks

Why my CSS is not working properly?
I have tried it through live server, Its alright but after committing and the server is live there is a design that I have uploader previously.

**Reply by pro-hacker18** [reply]
thank you it was  a big proplem for me

**Comment by noethix** [comment]
<p>Thank you <a class="mention" href="/sethclydesdale">@sethclydesdale</a> for your detailed answer and the tips ! But it didn’t work either…</p>

**Comment by lee-dohm** [comment]
<p>CSS appears to be working when I look at <a href="https://noethix.github.io/trial" rel="nofollow noopener">https://noethix.github.io/trial</a>. Do you see it as working now? Or can you describe what is not working for you?</p>

**Comment by Pr0fiLeRR** [comment]
<p>Thank you, that helped me so much; not only for CSS but also for IMAGE directories (I used to work with underline before its names, like: “_css or _image”).</p>
<p>So, I had to took them off and changed on HTML and CSS as well so it could work.</p>

**Comment by Fathima-Syeda** [comment]
<p>Hello, I am facing the same issue. Did you happen to find a fix for it?</p>
<blockquote>
<p><a class="mention" href="/noethix">@noethix</a> wrote:</p>
</blockquote>
<blockquote>
<p>Thank you <a class="mention" href="/sethclydesdale">@sethclydesdale</a> for your detailed answer and the tips ! But it didn’t work either…</p>
</blockquote>

**Reply by Manishak798** [reply]
Hey Fathima, you can consider checking the path you provided while uploading your files inside the repo, you need to upload same file and flowers as it is written in your local HTML file.

**Comment by SeemSach** [comment]
<p>Hi,</p>
<p>I tried few things which worked for me. I assume those are not major reasons; but still you can try</p>
<ol>
<li>
<p>Remove the css folder and keep your index file and css at same place, and remove the hierarchy in link attribute as href=“main.css”. </p>
</li>
<li>
<p>Add in head tag</p>
</li>
</ol>
<p>        &lt;meta charset=“UTF-8”&gt;<br>
        &lt;meta name=“viewport” content=“width=device-width, initial-scale=1.0”&gt;<br>
        &lt;meta http-equiv=“X-UA-Compatible” content=“ie=edge”&gt;<br>
and then &lt;link rel=“stylesheet” type=“text/css” href=“main.css”&gt;<br>
and then &lt;title&gt; &lt;/title&gt;<br>
try shuffling the order.</p>
<p>It worked then; I reshuffled the order to check if that is the reason; it is working since then for no absolute reason for shuffling the order.</p>
<p>And I am assuming each time you commit the change; in settings under GitHub Pages Section revisit the master brance in source option and then publish. </p>

**Reply by muhammed-py** [reply]
thank you, it worked for me, i only had to add the type="text/css"

**Reply by Jean-Laurore** [reply]
Thanks that work for me

**Comment by SeemSach** [comment]
<p>Hi,</p>
<p>I tried few things which worked for me. I assume those are not major reasons; but still you can try</p>
<ol>
<li>
<p>Remove the css folder and keep your index file and css at same place, and remove the hierarchy in link attribute as href=“main.css”. </p>
</li>
<li>
<p>Add in head tag</p>
</li>
</ol>
<p>        &lt;meta charset=“UTF-8”&gt;<br>
        &lt;meta name=“viewport” content=“width=device-width, initial-scale=1.0”&gt;<br>
        &lt;meta http-equiv=“X-UA-Compatible” content=“ie=edge”&gt;<br>
and then &lt;link rel=“stylesheet” type=“text/css” href=“main.css”&gt;<br>
and then &lt;title&gt; &lt;/title&gt;<br>
try shuffling the order.</p>
<p>It worked then; I reshuffled the order to check if that is the reason; it is working since then for no absolute reason for shuffling the order.</p>
<p>And I am assuming each time you commit the change; in settings under GitHub Pages Section revisit the master brance in source option and then publish. </p>

**Comment by MangbawiTawmbing** [comment]
<p>Can you help me with mine? </p>
<p><a href="https://github.com/MangbawiTawmbing/ludo" rel="nofollow noopener">https://github.com/MangbawiTawmbing/ludo</a></p>

**Comment by tmetz** [comment, marked as answer]
<p>Hello <a class="mention" href="/noethix">@noethix</a>,</p>
<p>Thanks for bringing up this topic, as this is a very common issue we see!  First of all, it does appear that after making the change suggested by <a class="mention" href="/SethClydesdale">@SethClydesdale</a> your CSS did start working, and <a href="https://noethix.github.io/trial/">https://noethix.github.io/trial/</a> looks great!  Sometimes it can take a minute for updates to show up, or an old version is cached by your browser, and I suspect that’s what happened here.</p>
<p>It also looks like <a class="mention" href="/mangbawitawmbing">@MangbawiTawmbing</a>’s issue has been resolved as well, as I see that <a href="https://mangbawitawmbing.github.io/ludo/dicee.html">https://mangbawitawmbing.github.io/ludo/dicee.html</a> is correctly displaying its CSS.</p>
<p>For all those coming to this topic later with a similar issue, below are some common situations where we see CSS not working properly, and how to fix it.</p>
<h2>Check for these common issues:</h2>
<h3 id="heading--relative-paths">Incorrect URL to the stylesheet</h3>
<p>Sometimes people accidentally use a stylesheet URL/directory path for their local computer, e.g. <code>&lt;link rel=“stylesheet” type=“text/css” href=“C:\Users\MyName\Project\styles.css”&gt;</code></p>
<p><strong>Solution:</strong><br>
Use a relative path to refer to your file, e.g. <code>&lt;link rel=“stylesheet” type=“text/css” href=“styles.css”&gt;</code>.</p>
<p><a href="https://developer.mozilla.org/en-US/docs/Learn/Common_questions/What_is_a_URL#Absolute_URLs_vs_relative_URLs">You can learn more about relative paths and how to create them here.</a></p>
<h3>Incorrect path to the stylesheet</h3>
<p>Sometimes people reference the stylesheet at the wrong place, e.g.</p>
<p><code>&lt;link rel=“stylesheet” type=“text/css” href=”css/styles.css”&gt;</code> but the stylesheet is actually at <code>&lt;link rel=“stylesheet” type=“text/css” href=“styles.css”&gt;</code>.</p>
<p><strong>Solution:</strong><br>
Double check the path to the stylesheet.</p>
<h3>Incorrect capitalization</h3>
<p>People can also accidentally have capitalization that doesn’t match the filename. For example, the stylesheet is referenced like this:<br>
<code>&lt;link rel=“stylesheet” type=“text/css” href=“styles.css”&gt;</code>.</p>
<p>However, the file is named <code>styles.CSS</code>. As a result, the stylesheet won’t be loaded since since GitHub Pages is case-sensitive.</p>
<p><strong>Solution:</strong><br>
Double check to ensure all capitalization matches.</p>
<p><strong>Better Solution:</strong><br>
Use all lowercase letters for all filenames and URLs.</p>
<h3>Missing file or directory</h3>
<p>Sometimes we find that the stylesheet or even entire css directory was never uploaded or pushed to the repository.  This often happens when using the web interface to upload files.</p>
<p><strong>Solution:</strong><br>
Ensure all files and directories have been pushed to the repository, so that the contents of the GitHub repository matches the contents of your local repository.</p>
<h3>CSS stops working after adding custom domain</h3>
<p>We often have people write in to tell us that their CSS was working fine until they added a custom domain to their repository.  One way to prevent many of these types of issues is to <a href="#heading--relative-paths">use relative paths</a>, but some site builders do use the full URL as a base path, so you may need to change a setting or two.</p>
<p><strong>Solution:</strong><br>
Change React or Jekyll settings:</p>
<p>For Jekyll sites, edit <code>_config.yml</code>.  If values are set for <code>baseurl</code> or <code>url</code> variables, we recommend deleting these lines entirely. This will allow GitHub Pages to detect and configure the ideal <code>baseurl</code> and <code>url</code> values in the background, automatically adjusting for changes to your site’s custom domain.</p>
<p>For React sites, edit <code>package.json</code>.  Change the “homepage” setting to your custom domain. Ensure that this setting matches the exact domain specified in your repository’s <code>CNAME</code> file.</p>
<p>For example, if your <code>CNAME</code> file contains <code>mydomain.com</code>, your <code>package.json</code> file should contain:</p>
<pre><code class="lang-auto">“homepage”: “https://domain.com”
</code></pre>
<p>If your <code>CNAME</code> file contains <code>www.mydomain.com</code>, your <code>package.json</code> file should contain:</p>
<pre><code class="lang-auto">“homepage”: “https://www.domain.com”
</code></pre>
<h2>Additional Resources</h2>
<p>If you are here looking for a solution to your CSS issue, I hope we’ve been able to help with the information above!  Here are some additional resources that may be helpful:</p>
<p>Explanation of absolute vs. relative paths:<br>
</p><aside class="onebox whitelistedgeneric">
  <header class="source">
      <img src="https://user-images.githubusercontent.com/6455271/181071459-e8f4fe35-5839-43b8-b0b6-41eb21c78e8d.png" class="site-icon" width="32" height="32">
      <a href="https://developer.mozilla.org/en-US/docs/Learn/Common_questions/What_is_a_URL#Absolute_URLs_vs_relative_URLs" target="_blank">MDN Web Docs</a>
  </header>
  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/6455271/181071460-6f5898ab-7daa-4ab8-a008-2ada0542024f.png" class="thumbnail onebox-avatar" width="500" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/e/e5c522d667c53fbfb0ada5f535969c16f9748c84_2_500x500.png, https://aws1.discourse-cdn.com/github/original/2X/e/e5c522d667c53fbfb0ada5f535969c16f9748c84.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/e/e5c522d667c53fbfb0ada5f535969c16f9748c84.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/e/e5c522d667c53fbfb0ada5f535969c16f9748c84_2_10x10.png">

<h3><a href="https://developer.mozilla.org/en-US/docs/Learn/Common_questions/What_is_a_URL#Absolute_URLs_vs_relative_URLs" target="_blank">What is a URL?</a></h3>

<p>With Hypertext and HTTP, URL is one of the key concepts of the Web. It is the mechanism used by browsers to retrieve any published resource on the web.</p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>Introduction to using cascading style sheets:<br>
</p><aside class="onebox whitelistedgeneric">
  <header class="source">
      <img src="https://user-images.githubusercontent.com/6455271/181071461-51465df5-d098-491f-929d-0311bb9c09e0.png" class="site-icon" width="32" height="32">
      <a href="https://developer.mozilla.org/en-US/docs/Web/CSS" target="_blank">MDN Web Docs</a>
  </header>
  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/6455271/181071462-2ba57b11-bfe8-4c76-a289-c3087aafeaf4.png" class="thumbnail onebox-avatar" width="500" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/e/e5c522d667c53fbfb0ada5f535969c16f9748c84_2_500x500.png, https://aws1.discourse-cdn.com/github/original/2X/e/e5c522d667c53fbfb0ada5f535969c16f9748c84.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/e/e5c522d667c53fbfb0ada5f535969c16f9748c84.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/e/e5c522d667c53fbfb0ada5f535969c16f9748c84_2_10x10.png">

<h3><a href="https://developer.mozilla.org/en-US/docs/Web/CSS" target="_blank">CSS: Cascading Style Sheets</a></h3>

<p>Cascading Style Sheets (CSS) is a stylesheet language used to describe the presentation of a document written in HTML</p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>Instructions on deploying a React app to GitHub Pages:<br>
</p><aside class="onebox whitelistedgeneric">
  <header class="source">
      <img src="https://create-react-app.dev/img/favicon/favicon.ico" class="site-icon" width="64" height="64">
      <a href="https://create-react-app.dev/docs/deployment/#github-pages" target="_blank">create-react-app.dev</a>
  </header>
  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/362;"><img src="https://user-images.githubusercontent.com/6455271/181071465-ae1ac3bc-eb09-4211-bd08-06ea68a2b708.png" class="thumbnail" width="690" height="362" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/e/e9d52951b6799822784160951f379815293e6f9a_2_10x10.png"></div>

<h3><a href="https://create-react-app.dev/docs/deployment/#github-pages" target="_blank">Create React App · Set up a modern web app by running one command.</a></h3>

<p>`npm run build` creates a `build` directory with a production build of your app. Set up your favorite HTTP server so that a visitor to your site is served `index.html`, and requests to static paths like `/static/js/main..js` are served with the...</p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

**Reply by wh33les** [reply]
Hi, I am having the problem with CSS not rendering after I switched to a custom domain.  I tried using the absolute path (href="style.css", though my style file is actually scss).  I got rid of the url and baseurl lines in my _config file.  Things were working fine several commits ago, so I reverted to a commit where things did work and they don't work now.  Is there anything else I can try?

**Reply by wh33les** [reply]
My site's repository is https://github.com/wh33les/codingblog.

**Comment by HeySkidee** [comment]
<p>Hello :slight_smile: ,</p>
<p>i also encountered same problem . i tried diff ways and methods but still the css was not working in <em><strong>Github Pages</strong></em></p>
<p><strong>[SOLUTION]</strong> : first make your complete project locally on your PC and after finishing it. create a repository and click on the option <strong>uploading an existing file</strong> :</p>
<p><img src="https://user-images.githubusercontent.com/59761032/181071468-ab4a096a-a453-4d53-ab7d-9b0d8e6994d6.png" alt="1" data-base62-sha1="pMGoYlOasssjJXHfNVxgoBY25mg" width="610" height="268"></p>
<hr>
<p>and then just drag and drop your whole project. if u want to edit something, then directly edit the code on github website. don’t use command line for github pages.</p>
<p>my repo : <a href="https://github.com/HarshitKumarOjha/HarshitKumarOjha.github.io" rel="nofollow noopener">https://github.com/HarshitKumarOjha/HarshitKumarOjha.github.io</a><br>
Result : <a href="https://harshitkumarojha.github.io/" rel="nofollow noopener">https://harshitkumarojha.github.io/</a></p>

**Reply by ruestitch** [reply]
Thanks for the suggestion, @HeySkidee. I'm having the same issue with site rendering after checking the advice provided. 

When I tried the solution you've suggested, I got the "too many files" error for a simple site. Does uploading the files in batches have any impact?

**Comment by amanjotsahni** [comment]
<p>your solution actually worked. thank you so much</p>

**Comment by gradientus** [comment]
<p>This did work and it is such a strange workaround.<br>
In addition, other strange behavior I noticed:<br>
css/style.css didn’t work.  ./css/style.css didn’t work.<br>
I had to put style.css in the root directory.  I don’t have a problem with it being there but it seem GitHub needs to do some bug triage.</p>
<p>Adding image path in the css with ./img/1.jpg didn’t work., neither did img/1.jpg.<br>
I ended up adding the image to the root as well as another workaround.</p>
<p>Not the greatest behavior.  There definitely is a bug here.</p>
<p>You can see the oddities in <a href="https://gradientus.github.io/tea/" rel="nofollow noopener">https://gradientus.github.io/tea/</a></p>

**Comment by Muskan-Malhotra** [comment]
<p>Thank you it really helped a lot</p>

**Comment by Rohit-1604** [comment]
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/2/2e01eea9d5bb5464a3e12f3f0ece65b815c36dc9.png" data-download-href="/uploads/short-url/6z06pXVet1nQoIVrHwwrcsU6f57.png?dl=1" title="Screenshot (52)" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/64626105/181071475-e304cf64-8aee-4867-b5dd-8625bc8c5535.png" alt="Screenshot (52)" data-base62-sha1="6z06pXVet1nQoIVrHwwrcsU6f57" width="690" height="321" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/2/2e01eea9d5bb5464a3e12f3f0ece65b815c36dc9_2_690x321.png, https://aws1.discourse-cdn.com/github/optimized/2X/2/2e01eea9d5bb5464a3e12f3f0ece65b815c36dc9_2_1035x481.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/2/2e01eea9d5bb5464a3e12f3f0ece65b815c36dc9_2_1380x642.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/2/2e01eea9d5bb5464a3e12f3f0ece65b815c36dc9_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screenshot (52)</span><span class="informations">1913×890 65.8 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>my bootstrap code is not working when i uploaded my file on github</p>
<aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">
      <a href="https://github.com/Rohit-1604/fitness" target="_blank" rel="nofollow noopener">GitHub</a>
  </header>
  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/64626105/181071476-71cc7f9a-7de5-44c5-afb7-84eaf6a52328.png" class="thumbnail onebox-avatar" width="420" height="420">

<h3><a href="https://github.com/Rohit-1604/fitness" target="_blank" rel="nofollow noopener">Rohit-1604/fitness</a></h3>

<p>Contribute to Rohit-1604/fitness development by creating an account on GitHub.</p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

**Comment by tmetz** [comment]
<p>Hi <a class="mention" href="/rohit-1604">@Rohit-1604</a>!  Jekyll is ignoring the <code>node_modules</code> directory, which is why anything in it (like Bootstrap’s CSS and JS) is not being found.</p>
<p>Since it doesn’t look like you’re using Jekyll, one way to solve this would be to create an empty file named <code>.nojekyll</code> in the root of your publishing source.  This will bypass the Jekyll portion of the Pages build process, and you should be able to successfully call files that are within <code>node_modules</code>.</p>

**Reply by ikukona** [reply]
holy f you made my day thank you so so so much

**Reply by kingptvoi** [reply]
dayum bro i'm late but thank you dude! your aproach was the only one that worked for me :P

**Reply by tojumikie** [reply]
Wow this worked for me, after fighting this thing for a whole month!
I was just about to give up and host my website on AWS or something else.
you made my day

**Reply by roberiof** [reply]
love u

**Reply by Firgrep** [reply]
my god, this fixed it for me as well. I had the issue that my astro build wasn't working in that the css and js files weren't being loaded despite path being correct. Once I moved these files manually out of `/_astro` and into root, edited the path in the index.html, everything worked, so something was messing up the access to that folder. Thanks supermuch!

**Comment by Rohit-1604** [comment]
<p>Now I have created the empty file named    .nojekyll  in the repository root. but even after that bootstrap is not applied on the page. root repository -   <a href="https://github.com/Rohit-1604/fitness.git" rel="nofollow noopener">https://github.com/Rohit-1604/fitness.git</a><br>
website repository - <a href="https://rohit-1604.github.io/fitness/" rel="nofollow noopener">https://rohit-1604.github.io/fitness/</a></p>

**Comment by ghost** [comment]
<p>Copy follwing code to your site <code>_config.yml</code></p>
<pre><code class="lang-auto">theme: null
</code></pre>
<p>Or create a new file <code>.nojekyll</code></p>

**Comment by joaomarcosribeiroc** [comment]
<p>I had this problem and I couldn’t find solution anywhere. All the folders where I placed css at were named with names starting with underscore( _ ), so I take off all underscores then it works. For example, the “_css” folder was renamed to “css”. I don’t know why it happens. Does GitHub Pages pages treat folders whose name starts with underscore as they are hidden, the same way linux hide folders whose name starts with a period( . )? Maybe</p>

**Reply by EkaterinaOssetchkina** [reply]
This fixed it for me! Thank you :)

**Comment by VaishnaviPandey** [comment]
<p>I am facing the same problem. HTML is working, but CSS is not being applied.I can’t figure it out kindly help. link to website is: <a href="https://vaishnavipandey.github.io/Personal-CV/" rel="nofollow noopener">https://vaishnavipandey.github.io/Personal-CV/</a></p>

**Reply by kingptvoi** [reply]
Hi! You could edit the _config.yml file content to `theme: null`
_**or**_
add a .nojekyll file, without any content in it in your /root folder

**Comment by xinwei-zhuang** [comment]
My css file doesn't work properly and the above debugging doesnt help...
https://xinwei-zhuang.github.io/
Attached image from my local folder which doesn't appear the same in the github webpage..
![WeChat Screenshot_20220901235034](https://user-images.githubusercontent.com/39757009/188222751-48cbf1ee-a981-4b31-a37a-fa2590f1a71c.png)

**Reply by Manishak798** [reply]
do check your folder or file path if you used external CSS.

**Comment by shaikaleem254** [comment]
I am facing the same problem. HTML is working, but CSS is not being applied.I can’t figure it out kindly help. link to website is: https://shaikaleem254.github.io/portfolio/

**Reply by NazmusSayad** [reply]
Update your css path

Your current code;
```
<link rel="stylesheet" type="text/css" href="./css/style.css">
```

Make this:
```
<link rel="stylesheet" type="text/css" href="./style.css">
```

**Comment by Lanarish** [comment]
Hi! I have the same problem, some bg-images don't show on the page after deployment. I tried all advice from this thread, removed all slashes from paths, add type="text/css" and even moved index.css to root, but it doesn't work for me. Maybe I'm blind :(
```class = "container__banner"``` dosen't show ```bg-image: home_first-block_bg.png``` on main page  and  ```class="header__content"``` dosen't show ```bg-image: template_header.jpg```on other pages. I've tried to change paths in URL for bg-image, but it doesn't work too. Although locally it's works!
main page locally
![image](https://github.com/community/community/assets/77628644/fafd8dd4-7ae9-44f7-89c6-c0c200f9eac7)
main page deployed
![image](https://github.com/community/community/assets/77628644/83e62f11-f649-4852-a42f-a6a8cdcd221d)

My repo - https://github.com/Lanarish/frontend66_group1
Pages - https://lanarish.github.io/frontend66_group1/

**Reply by ALTUS-15** [reply]
Telegram: @levangun2 
Senior on frontend and backend
Write

**Comment by divyasree-a** [comment]
Hi! I had the same problem. I'm coding using the Angular framework and had to build and deploy the website, so making changes in CSS messed with the build causing the website to remain the same no matter what I attempted to do.

So, anyone else having the same issue and using Angular, please rebuild and redeploy your website.

**Comment by leventkeles** [comment]
I just had a similar problem, the names of my CSS and JS files started with _. Normally it works without any problems, but such a problem occurred on Github.io. As a solution, I fixed it by deleting the _ character at the beginning of the file names. Also, do not forget to change the href link after changing the file name.

**Reply by pabloamayab** [reply]
You can solve this problem by simply adding the .nojekyll file to the root of your repository.
For more information: https://github.com/orgs/community/discussions/23166

**Reply by Shivshankar-kumar** [reply]
You Can check your file name. And put all file in same folder whether image, css or any, but not should be sub- folder.

It is working https://shivshankar-kumar.github.io/port/

**Comment by SUSHREE-0312** [comment]
Hi .. @SethClydesdale  could you please help me I deployed my portfolio and the CSS is not applied. this is my link.     [https://sushree-0312.github.io/Personal-Portfolio/](url)

**Comment by rilwanhaq** [comment]
https://rilwanhaq.github.io/


why my css is not applied.

**Reply by wh33les** [reply]
Your CSS seems to render now.  Can you tell me what you did to fix the problem?

**Comment by hunterandtheaxe** [comment]
I'm having this issue right now. Can someone help me with this?
https://github.com/hunterandtheaxe/hunterandtheaxe.github.io.git

**Reply by hunterandtheaxe** [reply]
found the issue. My CSS folder is in uppercase while I linked the folder on the html file in lower case.

**Comment by hunterandtheaxe** [comment]
My issue was that when I linked the css file in my index.html I had a
capitalization issue in the style sheet’s file name
<link rel=“stylesheet” href”css/style.css”>
where it should’ve been
<link rel=“stylesheet href”CSS/style.css”>

For your repo I believe you might have not included this folder “css” along
with the file “main.css” in your repo because I can’t find it anywhere.

<link rel=“stylesheet” href=“{{ “/css/main.css” | prepend: site.baseurl }}”>

I might be wrong though because I linked the stylesheet in the index.html
whereas you linked it in your head.html file.

I might not be where you’re at with coding too but you might have some
issues with linking your .html files with your index.html because I’m not
seeing any sort of <link>’s that reference them in your index.html file

**Reply by wh33les** [reply]
Thanks for your reply!  I didn't realize there was a <link> in my head.html file, I've always been looking at my default.html file.  That very much could be the issue because I don't have a folder called "css" nor a file called "main.css".  I got some other suggestions for things to try so I will definitely post again if I find the solution.

**Comment by hunterandtheaxe** [comment]
Oh yeah of course, I’ll be on the lookout to see if your css ends up being
applied.

**Comment by LuigiFerraioli** [comment]
Even though the assets were correctly linked in the HTML files (e.g., `../css/style.css`), GitHub Pages did not serve them properly because it only publishes content from the root or the `/docs` folder.

To solve this, I added an `index.html` file in the root directory that redirects to the actual homepage located in `/docs/home.html`.

### New structure:

```
├── css/
│   └──style.css
├── docs/
│   ├── home.html
│   └── nextpage.html
├── image/
├── js/
└── index.html
```

Contents of `index.html` (at the root):

`<meta http-equiv="refresh" content="0; url=docs/home.html" />`
